packages/astro/src/vite-plugin-build-html/index.ts

```diff
@@ -21,7 +21,7 @@ const ASTRO_PAGE_PREFIX = '@astro-page';
 const ASTRO_SCRIPT_PREFIX = '@astro-script';
 
 const ASTRO_EMPTY = '@astro-empty';
-
+const STATUS_CODE_RE = /^\d{3}$/;
```

Comment:
Afraid this might catch too much. A user *might* have a 3 digit number as a regular page in their site. Would feel better if we just captured the 404 and 500 (anything else?) since that's what providers often support.

Reply:
+1 to just treating `404.astro` as a special case. I believe Vercel and Netlify actually use this as the generic error page for all errors.

Reply:
Yep, will do this!